pycqed/instrument_drivers/physical_instruments/ZurichInstruments/UHFQuantumController.py

```diff
@@ -199,24 +214,89 @@ def __init__(self,
+        self._dio_calibration_delay = 0
+
+        # Define parameters that should not be part of the snapshot
+        self._params_to_exclude = set(['features_code', 'system_fwlog', 'system_fwlogenable'])
+
         # Our base class includes all the functionality needed to initialize the parameters
         # of the object. Those parameters are read from instrument-specific JSON files stored
         # in the zi_parameter_files folder.
-        super().__init__(name=name, device=device, interface=interface, server=server, port=port, **kw)
+        super().__init__(name=name, device=device, interface=interface,
+                         server=server, port=port, num_codewords=2**nr_integration_channels,
+                         **kw)
+
+        # Disable disfunctional parameters from snapshot
+        self._params_to_exclude = set(['features_code', 'system_fwlog', 'system_fwlogenable'])
 
         # Set default waveform length to 20 ns at 1.8 GSa/s
         self._default_waveform_length = 32
 
+        # Mask used for detecting codeword activity during DIO calibration
+        self._dio_calibration_mask = None
+
         t1 = time.time()
-        print('Initialized UHFQC', self.devname, 'in %.2fs' % (t1-t0))
+        log.info(f'{self.devname}: Initialized UHFQC in {t1 - t0}s')
 
     ##########################################################################
-    # Private methods
+    # Overriding Qcodes InstrumentBase methods
+    ##########################################################################
+
+    def snapshot_base(self, update: bool=False,
+                      params_to_skip_update =None,
+                      params_to_exclude = None ):
+        """
+        State of the instrument as a JSON-compatible dict.
+        Args:
+            update: If True, update the state by querying the
+                instrument. If False, just use the latest values in memory.
+            params_to_skip_update: List of parameter names that will be skipped
+                in update even if update is True. This is useful if you have
+                parameters that are slow to update but can be updated in a
+                different way (as in the qdac)
+        Returns:
+            dict: base snapshot
+        """
+
+        if params_to_exclude is None:
+            params_to_exclude = self._params_to_exclude
+
+        snap = {
+            "functions": {name: func.snapshot(update=update)
+                          for name, func in self.functions.items()},
+            "submodules": {name: subm.snapshot(update=update)
+                           for name, subm in self.submodules.items()},
+            "__class__": full_class(self)
+        }
+
+        snap['parameters'] = {}
+        for name, param in self.parameters.items():
+            if params_to_exclude and name in params_to_exclude:
+                pass
+            elif params_to_skip_update and name in params_to_skip_update:
+                update_par = False
```

Comment:
This parameter does not seem to have any effect.

Reply:
If it does not hurt, let's just keep it :-). We need to stay in sync with the DCL version as closely as possible in order to enable further driver updates in the future. I would say:
- If we spot things in their code that are definitely wrong and have a negative effect, we can create pull requests in their git.
- If we spot things that we definitely need differently, but that look fine for them, we have to change them in our fork (and live with the trouble in future merges).
- If we spot things that do not really hurt, let's be pragmatic and just leave them as they are for the sake of allowing easy merging (same for code formatting!).

Reply:
Agree!